test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManagerSpec.scala

```diff
@@ -0,0 +1,208 @@
+
+  "A Tax Enrolment Retry Manager actor" must {
+
+    "send a message to the mongo actor if tax enrolnment call was successful" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+
+      val taxEnrolmentRequest = TaxEnrolmentRequest(
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        2,
+        0,
+        249237542,
+        LocalDateTime.now()
+      )
+
+      app ! MarkTaxEnrolmentRetryRequestAsSuccess(taxEnrolmentRequest)
+
+      mongoProbe.receiveN(2)
+    }
+
+    "send a message to the mongo actor if the elapsed time has exceeded the max elapsed time" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+
+      val taxEnrolmentRequest = TaxEnrolmentRequest(
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        2,
+        0,
+        249237542,
+        LocalDateTime.now()
+      )
+
+      app ! RetryTaxEnrolmentRequest(taxEnrolmentRequest)
+      mongoProbe.receiveN(2)
+    }
+
+    "send a message to the connector actor if the elapsed time has exceeded the max elapsed time" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
+          )
+        )
+      )
+
+      val taxEnrolmentRequest = TaxEnrolmentRequest(
+        "userId",
+        "someref",
+        Address.UkAddress("line1", None, None, None, "KO"),
+        "InProgress",
+        0,
+        0,
+        0,
+        LocalDateTime.of(2019, 12, 1, 12, 10)
+      )
+
+      app ! RetryTaxEnrolmentRequest(taxEnrolmentRequest)
+      // need to advance the time
+//      time.advance(FiniteDuration(10, TimeUnit.SECONDS))
+
```

Comment:
The time.advance does not seem to advance the time. Can we take a look together on this one